File: bump-ios-build.ts
#!/usr/bin/env tsx
import { execSync } from "child_process";

const plistPath = "ios/Timesheet/Info.plist"; // adjust if needed

function run(cmd: string): string {
  return execSync(cmd, { stdio: "pipe" }).toString().trim();
}

function getValue(key: string): string {
  return run(`/usr/libexec/PlistBuddy -c "Print ${key}" "${plistPath}"`);
}

function setValue(key: string, value: string | number) {
  run(`/usr/libexec/PlistBuddy -c "Set ${key} ${value}" "${plistPath}"`);
}

function assertCleanGit() {
  const status = run("git status --porcelain");
  if (status.length > 0) {
    console.error("❌ Working directory is not clean. Please commit or stash your changes first.");
    process.exit(1);
  }
}

try {
  // 🚫 Stop if there are uncommitted changes
  assertCleanGit();

  const version = getValue("CFBundleShortVersionString");
  const build = parseInt(getValue("CFBundleVersion"), 10) + 1;

  setValue("CFBundleVersion", build);

  const tag = `ios.${version}.${build}`;
  const commitMsg = tag;

  run(`git add "${plistPath}"`);
  run(`git commit -m "${commitMsg}"`);
  run(`git tag "${tag}"`);

  console.log(`✅ Updated build number to ${build}, committed and tagged as ${tag}`);
} catch (e: any) {
  console.error("❌ Error:", e.message);
  process.exit(1);
}

File: bump_ios_build.java
///usr/bin/env jbang --java 25 "$0" "$@" ; exit $?
//SOURCES utils/**.java

import java.io.IOException;

import utils.GitProxy.CommitSource;

import static utils.GitProxy.assertCleanGit;
import static utils.GitProxy.commit;
import static utils.Shell.run;
import static utils.TextFormat.bold;

String plistPath = "ios/Timesheet/Info.plist";

void main() {
    try {
        assertCleanGit();

        var version = getValue("CFBundleShortVersionString");
        var build = getValue("CFBundleVersion");
        var nextBuild = String.valueOf(Integer.parseInt(build) + 1);

        IO.println(String.format("v%s %s -> %s", version, bold(build), bold(nextBuild)));

        setValue("CFBundleVersion", nextBuild);

        commit(CommitSource.IOS, version, nextBuild);
    } catch (Exception e) {
        System.err.println("❌ Error: " + e);
        System.exit(1);
    }
}

String getValue(String key) throws IOException {
    return run(String.format("/usr/libexec/PlistBuddy -c 'Print %s' '%s'", key, plistPath));
}

void setValue(String key, String value) throws IOException {
    run(String.format("/usr/libexec/PlistBuddy -c 'Set %s %s' '%s'", key, value, plistPath));
}
